src/omnicore/notifications.cpp

```diff
@@ -19,8 +20,41 @@
 namespace mastercore
 {
 
-//! Stores alert messages for Omni Core
-static std::string global_alert_message;
+//! Vector of currently active Omni alerts
+std::vector<AlertData> currentOmniAlerts;
+
+/**
+ * Adds a new alert to the alerts vector
+ *
+ */
+void AddAlert(uint16_t alertType, uint32_t alertExpiry, const std::string& alertMessage)
+{
+    AlertData newAlert;
+    newAlert.alert_type = alertType;
+    newAlert.alert_expiry = alertExpiry;
+    newAlert.alert_message = alertMessage;
+
+    // very basic sanity checks to catch malformed packets
```

Comment:
Hm... I'm not really a fan of more magic numbers here. For example, do the timestamps also work on testnet or regtest mode?

While a non-existing alert type should be rejected, and malformed packages in general, I'm not really sure, if we *need* further constraints.

Another example: we're (arguably) off in the context of the client version as number, and https://github.com/OmniLayer/omnicore/pull/162 updates the number to 1000000 (for 0.0.10), so that this release has a higher number than Master Core 0.0.9.3 (which has 900300), which would result in failed checks here, only because we tried to make guesses about the future (or made a mistake, like with the number, on the way).